iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/MultiplexingClientTests.java

```diff
@@ -1517,35 +1517,43 @@ public void disableDeviceAfterOpenAndAfterRegistration() throws Exception
         registryManager.updateDevice(deviceToDisable);
 
-        // verify that the disabled device loses its device session
-        long startTime = System.currentTimeMillis();
-        while (!connectionStatusChangeTrackers[0].wentDisconnectedRetrying)
+        try
         {
-            Thread.sleep(200);
-
-            if (System.currentTimeMillis() - startTime > FAULT_INJECTION_TIMEOUT_MILLIS)
+            // verify that the disabled device loses its device session
+            long startTime = System.currentTimeMillis();
+            while (!connectionStatusChangeTrackers[0].wentDisconnectedRetrying)
             {
-                fail("Timed out waiting for the disabled device's client to report DISCONNECTED_RETRYING");
+                Thread.sleep(200);
+
+                if (System.currentTimeMillis() - startTime > FAULT_INJECTION_TIMEOUT_MILLIS)
+                {
+                    fail("Timed out waiting for the disabled device's client to report DISCONNECTED_RETRYING");
+                }
             }
-        }
 
-        assertFalse(connectionStatusChangeTrackers[0].isOpen);
+            assertFalse(connectionStatusChangeTrackers[0].isOpen);
 
-        // Verify that the other devices on the multiplexed connection were unaffected
-        for (int i = 1; i < DEVICE_MULTIPLEX_COUNT; i++)
-        {
-            assertFalse(connectionStatusChangeTrackers[i].wentDisconnectedRetrying);
-            assertTrue(connectionStatusChangeTrackers[i].isOpen);
-        }
+            // Verify that the other devices on the multiplexed connection were unaffected
+            for (int i = 1; i < DEVICE_MULTIPLEX_COUNT; i++)
+            {
+                assertFalse(connectionStatusChangeTrackers[i].wentDisconnectedRetrying);
+                assertTrue(connectionStatusChangeTrackers[i].isOpen);
+            }
 
-        // Verify that the multiplexed connection itself was unaffected
-        assertFalse(multiplexedConnectionStatusChangeTracker.wentDisconnectedRetrying);
-        assertTrue(multiplexedConnectionStatusChangeTracker.isOpen);
+            // Verify that the multiplexed connection itself was unaffected
+            assertFalse(multiplexedConnectionStatusChangeTracker.wentDisconnectedRetrying);
+            assertTrue(multiplexedConnectionStatusChangeTracker.isOpen);
 
-        // Verify that the other devices can still send telemetry
-        testSendingMessagesFromMultiplexedClients(testInstance.deviceClientArray.subList(1, DEVICE_MULTIPLEX_COUNT));
+            // Verify that the other devices can still send telemetry
+            testSendingMessagesFromMultiplexedClients(testInstance.deviceClientArray.subList(1, DEVICE_MULTIPLEX_COUNT));
 
-        testInstance.multiplexingClient.close();
+            testInstance.multiplexingClient.close();
+        }
+        finally
+        {
+            deviceToDisable.setStatus(DeviceStatus.Enabled); // re enable the device in case it gets recycled
```

Comment:
Is this to protect against the case where the test throws and exception and the device it is using needs to be recycled?

Reply:
Is it possible to do this by default in disposeTestIdentity in the Tools instead? Unless that won't be called on exception. There is a lot to understand so asking you the question :)

Reply:
Typically tests don't disable their own devices at any point, so this try/finally logic is just specific to the tests that do disable devices at this point